lib/doctest.js

```diff
@@ -197,20 +212,46 @@ function processLine(
     } else if (trimmedLine === options.closingDelimiter) {
       accum.state = CLOSED;
     } else if (trimmedLine.charAt (0) === '>') {
+      accum.state = INPUT;
       value = stripLeading (1, ' ', stripLeading (1, '>', trimmedLine));
       accum.tests.push ($test = {});
-      $test[accum.state = INPUT] = {value: value};
+      $test[INPUT] = {value: value};
+      $test[OUTPUT] = {values: []};
       input ($test);
-    } else if (trimmedLine.charAt (0) === '.') {
+    } else if (accum.state === INPUT && trimmedLine.charAt (0) === '.') {
+      value = stripLeading (1, ' ', stripLeading (Infinity, '.', trimmedLine));
+      $test = accum.tests[accum.tests.length - 1];
+      $test[INPUT].value += '\n' + value;
+      appendToInput ($test);
+    } else if ((accum.state === OUTPUT || accum.state === LOG) &&
+               (trimmedLine.charAt (0) === '.')) {
       value = stripLeading (1, ' ', stripLeading (Infinity, '.', trimmedLine));
       $test = accum.tests[accum.tests.length - 1];
-      $test[accum.state].value += '\n' + value;
-      (accum.state === INPUT ? appendToInput : appendToOutput) ($test);
-    } else if (accum.state === INPUT) {
+      $test[OUTPUT].values[$test[OUTPUT].values.length - 1].value += '\n'
+                                                                  + value;
+      appendToOutput ($test);
+    } else if (MATCH_LOG.test (trimmedLine)) {
+      accum.state = LOG;
+      value = stripLeading (1, ' ', trimmedLine.replace (MATCH_LOG, ''));
+      $test = accum.tests[accum.tests.length - 1];
+      $test[OUTPUT].values.push ({
+        channel: MATCH_LOG.exec (trimmedLine)[1],
```

Comment:
I'm not totally happy with this (guarded) unsafe expression. I would prefer this approach:

```javascript
var match;
if (...) {
  // ...
} else if ((match = MATCH_LOG.exec (trimmedLine)) != null) {
  // ...
  // match[1]
  // ...
}
```